code/controlconfig/controlsconfigcommon.cpp

```diff
@@ -1523,126 +1253,489 @@ size_t find_control_by_text(SCP_string& text) {
+size_t read_bind_1(CC_preset &preset) {
+	CCB* item = nullptr;
+	SCP_string szTempBuffer;
+	int item_id = 0;
+
+	// $Bind:
+	stuff_string(szTempBuffer, F_NAME);
+	item_id = ActionToVal(szTempBuffer.c_str());
+
+	if (item_id >= 0) {
+		item = &preset.bindings[item_id];
+
+	} else {
+		// Control wasn't found
+		error_display(0, "Unknown Bind: %s\n", szTempBuffer.c_str());
+
+		return Control_config.size();
+	}
+	
+	if (optional_string("$Primary:")) {
+		CID cid = CID_NONE;
+		char flags = '\0';
+		short btn = 0;
+		if (required_string("$Controller:")) {
+			stuff_string(szTempBuffer, F_NAME);
+			cid = CIDToVal(szTempBuffer.c_str());
+		}
+		
+		// These items are required if the controller is defined
+		if (cid != CID_NONE) {
+			if (required_string("$Flags:")) {
+				stuff_CCF(flags, item_id);
+			}
+
+			if (required_string("$Input:")) {
 				stuff_string(szTempBuffer, F_NAME);
-				try {
-					key = mKeyNameToVal.at(szTempBuffer);
-				} catch (std::out_of_range&) {
-					Warning(LOCATION,"Table tried to set '$Key Default' for a key that doesn't exist: %s",szTempBuffer.c_str());
-				}
+				btn = InputToVal(cid, szTempBuffer.c_str());
 			}
 		}
 
-		if (optional_string("$Key Mod Shift:")) {
-			stuff_int(&iTemp);
-			key |= (iTemp == 1) ? KEY_SHIFTED : 0;
+		item->first.take(cid, btn, flags);
+	}
+
+	// Second verse, same as the first
+	if (optional_string("$Secondary:")) {
+		CID cid = CID_NONE;
+		char flags = '\0';
+		short btn = 0;
+		if (required_string("$Controller:")) {
+			stuff_string(szTempBuffer, F_NAME);
+			cid = CIDToVal(szTempBuffer.c_str());
 		}
 
-		if (optional_string("$Key Mod Alt:")) {
-			stuff_int(&iTemp);
-			key |= (iTemp == 1) ? KEY_ALTED : 0;
+		// These items are required if the controller is defined
+		if (cid != CID_NONE) {
+			if (required_string("$Flags:")) {
+				stuff_CCF(flags, item_id);
+			}
+
+			if (required_string("$Input:")) {
+				stuff_string(szTempBuffer, F_NAME);
+				btn = InputToVal(cid, szTempBuffer.c_str());
+			}
 		}
 
-		if (optional_string("$Key Mod Ctrl:")) {
-			stuff_int(&iTemp);
-			key |= (iTemp == 1) ? KEY_CTRLED : 0;
+		item->second.take(cid, btn, flags);
+	}
+
+	return static_cast<size_t>(item_id);
+}
+
+/**
+ * @brief Reads a section in controlconfigdefaults.tbl.
+ *
+ * @param[in] s Value of a call to optional_string_either(); 0 = "ControlConfigOverride" 1 = "ControlConfigPreset"
+ * @param[in] first_override Legacy support for unnamed #ControlConfigOverrides.  If this is the first unnamed
+ *   override, then overwrite the default preset, else, save it as an "<unnamed preset>"
+ * 
+ * @details ControlConfigPresets are read in the exact same manner as ControlConfigOverrides, however only the bindings are available for modification.
+ *  There may be only one #Override section, since it is in charge of non-binding members of the Control_config items
+ */
+void control_config_common_read_section(int s, bool first_override) {
+	Assert((s == 0) || (s == 1));
```

Comment:
This should be an assertion that dumps what s is. Otherwise the coder debugging is going to shrug if they can't reproduce.